core/service.go

```diff
@@ -93,19 +106,46 @@ func (s *Service) ProcessTransaction(e types.Extrinsic) error {
 
 // ProcessBlock attempts to add a block to the chain by calling `core_execute_block`
 // if the block is validated, it is stored in the block DB and becomes part of the canonical chain
-func (s *Service) ProcessBlock(b *types.BlockHeader) error {
-	return nil
+func (s *Service) ProcessBlock(b *types.Block) error {
```

Comment:
I'm inclined to believe these should be part of babe not service, thoughts?

Reply:
I think it should be part of service since ProcessBlock isn't BABE specific, it just executes a block through the runtime to update state